### PR TITLE
Caching results of insertLineNumbers and extractRangeByLineNumbers

### PR DESCRIPTION
**Pull request by CatoTH**
The first, simpler part of the performance improvements

### CODE REVIEW

File: openslides/motions/static/js/motions/diff.js [hunk 1]
```diff
@@ -4,11 +4,13 @@
 
 angular.module('OpenSlidesApp.motions.diff', ['OpenSlidesApp.motions.lineNumbering'])
 
-.service('diffService', function (lineNumberingService) {
+.service('diffService', function (lineNumberingService, $cacheFactory) {
```
Comment (normanjaeckel) [inline openslides/motions/static/js/motions/diff.js]: Ah I think it is necessary to use the array notation:
```javascript
.service('diffService', [
    'lineNumberingService',
    '$cacheFactory',
    function (lineNumberingService, $cacheFactory) {
        ...
    }
])
```
This is required if we later want to compress and uglify the code via gulp.

File: openslides/motions/static/js/motions/linenumbering.js [hunk 2]
```diff
@@ -14,14 +14,25 @@ angular.module('OpenSlidesApp.motions.lineNumbering', [])
  * No constructs like <a...><div></div></a> are allowed. CSS-attributes like 'display: block' are ignored.
  */
 
-.service('lineNumberingService', function () {
+.service('lineNumberingService', function ($cacheFactory) {
```
Comment (normanjaeckel) [inline openslides/motions/static/js/motions/linenumbering.js]: Here the same.

**Comment by thoessl** [comment]
@normanjaeckel Like in the second commit now? If it's ok like this, I'll squash those two commits.

**Comment by normanjaeckel** [comment]
Looks good.

**Comment by emanuelschuetze** [comment]
@CatoTH Please rebase. Then I'll merge.

**Pull request merged**